x/gamm/keeper/pool_service.go

```diff
@@ -62,6 +62,9 @@ func (k Keeper) CalculateSpotPrice(
 // - Sets bank metadata for the LP denom
 // - Records total liquidity increase
 // - Calls the AfterPoolCreated hook
+//
+// After the gammKeeper set pool to the state, this function calls the AfterPoolCreated hook
+// which is responsible for createing a gauge for each pool’s lockable duration
```

Comment:
Isn't this documented right above?

The fact that it creates gauges for each pool's lockable duration isn't the problem of the gamm module, thats the advantage of hooks! 

```suggestion
// - Calls the AfterPoolCreated hook (see app module wiring for what may be running on this)
```